### DISCUSSION TITLE
How to get state of automation triggers?

### DISCUSSION THREAD

**Discussion by DennisGaida** [zachowj/node-red-contrib-home-assistant-websocket#939] [Q&A] [answered]
I am using Home Assistant MQTT device triggers such as explained here: https://www.home-assistant.io/integrations/device_trigger.mqtt/

I can create HA-automations using these triggers, i.e. I can select these MQTT device triggers when creating/modifying a HA automation and it works flawlessly.

I don't know how to "catch" these **automation triggers** in Node-Red using this integration. The state of the entity doesn't seem to change. I also don't quite understand these automation triggers, just know that they work in HA and are the "best practice" to use right now. I actually tried finding the event using the **events: all** action, but that only showed my automation that uses the trigger, not the trigger itself.

Is there any way to use or read these HA automation triggers from Node-RED with this integration?

**Comment by zachowj** [comment]
https://www.home-assistant.io/integrations/device_trigger.mqtt/#left-arrow-click-configuration

Using the above example wouldn't you use an MQTT node to listen on `zigbee2mqtt/0x90fd9ffffedf1266/action`?

**Reply by DennisGaida** [reply]
Oh yes, of course I could use an MQTT node in Node-Red. I‘m wondering whether I can make use of this integration - using homeassistant.

**Reply by zachowj** [reply, marked as answer]
If you install the [Node-RED companion component](https://github.com/zachowj/hass-node-red) in HA. You can use the device node for device triggers.

**Reply by DennisGaida** [reply]
**device** trigger! thank you! I had pretty much tried all actions like *events: all*, *events: state*, *current state*, *trigger: state*... haven't tried device.
